/**:
  ros__parameters:
    control_rate: 30.0
    delay_compensation_time: 0.17

    enable_smooth_stop: true
    enable_overshoot_emergency: true
    enable_slope_compensation: false

    # state transition
    drive_state_stop_dist: 0.5
    drive_state_offset_stop_dist: 1.0
    stopping_state_stop_dist: 0.49
    stopped_state_entry_vel: 0.1
    stopped_state_entry_acc: 0.1
    emergency_state_overshoot_stop_dist: 1.5
    emergency_state_traj_trans_dev: 3.0
    emergency_state_traj_rot_dev: 0.7

    # drive state
    kp: 1.0
    ki: 0.1
    kd: 0.0
    max_out: 1.0
    min_out: -1.0
    max_p_effort: 1.0
    min_p_effort: -1.0
    max_i_effort: 0.3
    min_i_effort: -0.3
    max_d_effort: 0.0
    min_d_effort: 0.0
    lpf_vel_error_gain: 0.9
    current_vel_threshold_pid_integration: 0.5

    # smooth stop state
    smooth_stop_max_strong_acc: -0.5
    smooth_stop_min_strong_acc: -1.0
    smooth_stop_weak_acc: -0.3
    smooth_stop_weak_stop_acc: -0.8
    smooth_stop_strong_stop_acc: -3.4
    smooth_stop_max_fast_vel: 0.5
    smooth_stop_min_running_vel: 0.01
    smooth_stop_min_running_acc: 0.01
    smooth_stop_weak_stop_time: 0.8
    smooth_stop_weak_stop_dist: -0.3
    smooth_stop_strong_stop_dist: -0.5

    # stopped state
    stopped_vel: 0.0
    stopped_acc: -3.4
    stopped_jerk: -5.0

    # emergency state
    emergency_vel: 0.0
    emergency_acc: -5.0
    emergency_jerk: -3.0

    # acceleration limit
    max_acc: 3.0
    min_acc: -5.0

    # jerk limit
    max_jerk: 2.0
    min_jerk: -5.0

    # pitch
    use_trajectory_for_pitch_calculation: false
    lpf_pitch_gain: 0.95
    max_pitch_rad: 0.1
    min_pitch_rad: -0.1

    # vehicle parameters
    vehicle:
      cg_to_front_m: 1.228
      cg_to_rear_m: 1.5618
      mass_kg: 2400.0
      mass_fl: 600.0
      mass_fr: 600.0
      mass_rl: 600.0
      mass_rr: 600.0
      cf: 155494.663
      cr: 155494.663